src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManagerBase.cs

```diff
@@ -267,6 +290,74 @@ public override Task RemoveFromGroupAsync(string connectionId, string groupName,
             return WriteAckableMessageAsync(message, cancellationToken);
         }
 
+#if NET7_0_OR_GREATER
+        public override async Task<T> InvokeConnectionAsync<T>(string connectionId, string methodName, object[] args, CancellationToken cancellationToken = default)
+        {
+            if (IsInvalidArgument(connectionId))
+            {
+                throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(connectionId));
+            }
+
+            if (IsInvalidArgument(methodName))
+            {
+                throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(methodName));
+            }
+            // globally distinct invocationId
+            // $"{connectionId}{_callerId}{_clientResults.GetNewInvocation()}";
+            var invocationId = $"{connectionId}-{_serverGUID}-{_clientResults.GetNewInvocation()}";
+            var task = _clientResults. AddInvocation<T>(connectionId, invocationId, cancellationToken);
+            try
+            {
+                var message = AppendMessageTracingId(new ClientInvocationMessage(invocationId, connectionId, _callerId, SerializeAllProtocols(methodName, args, invocationId)));
+                await WriteAsync(message);
+            }
+            catch (Exception)
+            {
+                _clientResults.RemoveInvocation(invocationId);
+                throw;
+            }
+
+            try
+            {
+                return await task;
+            }
+            catch
+            {
+                throw;
+            }
+        }
+
+        public override async Task SetConnectionResultAsync(string connectionId, CompletionMessage result)
+        {
+            if (IsInvalidArgument(connectionId))
+            {
+                throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(connectionId));
+            }
+            if (_clientConnectionManager.ClientConnections.TryGetValue(connectionId, out var clientConnectionContext))
+            {
+                // Current server is a route server.
+                // In order to inform original caller server with the completion result, send a ClientCompletionMessage to service and the service will route ClientCompletionMessage to the original caller server.
+                if (_clientResults.CheckRoutedInvocation(result.InvocationId))
+                {
+                    var protocol = clientConnectionContext.Protocol;
+                    // if connectionId correct?
+                    var message = AppendMessageTracingId(new ClientCompletionMessage(result.InvocationId, connectionId, _callerId, protocol, SerializeCompletionMessage(result)[protocol]));
```

Comment:
Here has the client protocol information.

Reply:
fixed with https://github.com/Azure/azure-signalr/pull/1681#discussion_r979472132